examples/static-xor-mlp/Main.hs

```diff
@@ -110,26 +108,41 @@ xor t = (1 - (1 - a) * (1 - b)) * (1 - (a * b))
   a = select @1 @0 t
   b = select @1 @1 t
 
-type Device = '( 'D.CUDA, 0)
+newtype Xor = Xor { iters :: Int }
 
-main :: IO ()
-main = do
-  let numIters = 100000
-      learningRate = 0.1
-  initModel <- A.sample (MLPSpec :: MLPSpec 2 1 4 'D.Float Device)
-  let initOptim = mkAdam 0 0.9 0.999 (flattenParameters initModel)
-  (trained, _) <- foldLoop (initModel, initOptim) numIters $ \(model, optim) i -> do
-    input <-
-      Torch.Typed.Tensor.toDType @D.Float
-      .   gt (Torch.Typed.Tensor.toDevice @Device (0.5 :: CPUTensor 'D.Float '[]))
-      <$> rand @'[256, 2] @'D.Float @Device
+instance ( KnownNat batchSize
+         , KnownDevice device
+         , RandDTypeIsValid device 'D.Float
+         , ComparisonDTypeIsValid device 'D.Float
+         ) => Dataset Xor (Tensor device 'D.Float '[batchSize, 2]) where
+  getBatch _ _ =  Torch.Typed.Tensor.toDType @D.Float .
+                  gt (Torch.Typed.Tensor.toDevice @device (0.5 :: CPUTensor 'D.Float '[]))
+                  <$> rand @'[batchSize, 2] @'D.Float @device
+  numIters = iters 
+
+type Device = '( 'D.CUDA, 0)
```

Comment:
do you have a GPU you can test this with?

Reply:
Yep.